### PR TITLE
Feature/task 1

### PR DESCRIPTION
**Pull request by JarekParal**

### CODE REVIEW

File: src/components/Board.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        },
+      ],
+    }));
+  };
+
+  _editItem = (uuid, text) => {
+    const updatedItems = [...this.state.items];
+    const pos = updatedItems.findIndex(item => item.id === uuid);
+    updatedItems[pos].text = text;
+    this.setState(() => ({
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/components/Board.jsx]: I tried to use `prevState.items.map()`. Is that right way? https://github.com/JarekParal/kentico-onboarding-js/commit/ac35ffa7cb1620d506beec023fa299dfdd11aecc

File: src/components/Board.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      items: updatedItems
+    }));
+  };
+
+  render() {
+    return (
+      <div>
+        <ul className="list-group">
+          {this.state.items.map((item, index) => (
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/components/Board.jsx]: https://github.com/JarekParal/kentico-onboarding-js/commit/b0217a867d3ca4a731ac5390f380cc7a9bb54452

File: src/components/EditItem.jsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: this.props.id,
+      text: this.props.text,
+    };
+  }
+
+  _textEdit = (e) => {
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/components/EditItem.jsx]: https://github.com/JarekParal/kentico-onboarding-js/commit/5e3697c4d6f06d4dd584da04130b9909896ef1fb

File: src/components/ShowItem.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const ShowItem = (props) => (
+  <div
+    role="presentation"
+    onClick={props.handlerClick}
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/components/ShowItem.jsx]: https://github.com/JarekParal/kentico-onboarding-js/commit/b429efa651beb573b9ea91f06cea6b837f0df160

File: src/components/ShowItem.jsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  >
+      {props.pos}. {props.text}
+  </div>
+);
+
+ShowItem.displayName = 'ShowItem';
+
+ShowItem.propTypes = {
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/components/ShowItem.jsx]: https://github.com/JarekParal/kentico-onboarding-js/commit/b429efa651beb573b9ea91f06cea6b837f0df160

File: src/components/EditItem.jsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    pos: PropTypes.number.isRequired,
+    finishEdit: PropTypes.func.isRequired,
+    onSave: PropTypes.func.isRequired,
+    onDelete: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: this.props.id,
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/components/EditItem.jsx]: https://github.com/JarekParal/kentico-onboarding-js/commit/fc846add8d7310f1f1944e46de4fa3741cf302b5

File: src/components/EditItem.jsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  static propTypes = {
+    id: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+    pos: PropTypes.number.isRequired,
+    finishEdit: PropTypes.func.isRequired,
+    onSave: PropTypes.func.isRequired,
+    onDelete: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/components/EditItem.jsx]: https://github.com/JarekParal/kentico-onboarding-js/commit/96328c6a7eb3611ddae9ca2bb7b8abe5c4114cc8

File: src/components/UuidGen.jsx [hunk 8]
```diff
@@ -0,0 +1,5 @@
+const uuidGenerator = require('uuid/v4');
```
[the remark this answers is not in the document]
Comment (JarekParal) [reply src/components/UuidGen.jsx]: https://github.com/JarekParal/kentico-onboarding-js/commit/9f56db51daf36eb03b8b98cb9a37d01de3cfec71

**Comment by JarekParal** [comment]
Ready to a new review.

File: src/components/Board.jsx [hunk 9]
```diff
@@ -1,72 +1,68 @@
...
+    ],
+  };
 
   _addItem = (text) => {
     this.setState(prevState => ({
       items: [
         ...prevState.items,
         {
-          'id': uuidGen(),
+          'id': uuidGenerator(),
```
Comment (vaclavm) [inline src/components/Board.jsx]: You do not need ' ' 😉

File: src/utils/UuidGenerator.jsx [hunk 10]
```diff
@@ -0,0 +1,3 @@
+import { v4 as generator } from 'uuid';
+
+export const uuidGenerator = () => generator();
```
Comment (vaclavm) [inline src/utils/UuidGenerator.jsx]: The file should be named with lower case, it is not a model/component/class.

Comment (vaclavm) [reply src/utils/UuidGenerator.jsx]: Also, please create new test file and add basic test for the generator (you can test if it generates uuid in expected format).

File: src/utils/UuidGenerator.jsx [hunk 11]
```diff
@@ -0,0 +1,3 @@
+import { v4 as generator } from 'uuid';
```
Comment (vaclavm) [inline src/utils/UuidGenerator.jsx]: FYI: To get rid of the warning notification, you can import like `import * as generator from 'uuid/v4'`
image

File: src/App.jsx [hunk 12]
```diff
@@ -1,10 +1,9 @@
 import './sticky-footer.css';
-import React, { PureComponent } from 'react';
+import React from 'react';
 import { Board } from './components/Board';
 
-export class App extends PureComponent {
-  render() {
-    return (
+export const App = () => {
```
Comment (vaclavm) [inline src/App.jsx]: { } and return can be removed

File: src/components/List.jsx [hunk 13]
```diff
@@ -1,46 +1,28 @@
...
-            <Board />
-          </div>
-        </div>
-      </div>
-    );
+      this.props.items
+        .map((item, index) => (
+          <Item
+            key={item.id}
+            id={item.id}
```
Comment (vaclavm) [inline src/components/List.jsx]: FYI: You can pass whole item to Item component

File: src/utils/UuidGenerator.jsx (hunk 10 above)
Comment (JarekParal) [reply src/utils/UuidGenerator.jsx]: https://github.com/JarekParal/kentico-onboarding-js/commit/e57acd3d33b9309f169f6f23300941f97a0fa4da

**Comment by JarekParal** [comment]
Finish

**Pull request merged**